### DISCUSSION TITLE
Tool-Assisted Speedrun (TAStudio, Savestates and Piano roll)

### DISCUSSION THREAD

**Discussion by p0008874** [ruffle-rs/ruffle#8404] [Ideas] [answered]
I know this required #895 because why not.

For more detail: https://en.wikipedia.org/wiki/Tool-assisted_speedrun

**Comment by ghost** [comment]
libTAS does exactly this. Ruffle is not a TASing emulator.

**Reply by p0008874** [reply]
Can Linux users know what is cross-platform?
You guys are always give the worst experience from non Linux users.

**Reply by ghost** [reply]
I do not understand what you are trying to say.

"You guys are worst from non Linux users."
I'm not sure if this is supposed to be a hateful statement, but ¯\_(ツ)_/¯ (thanks github for messing up my shrug)

**Reply by Futurama39** [reply]
libTAS is not portable. That is by design. This is because libTAS has to do a lot of operations close to the kernel, and that is only really doable if you actually know how the kernel works. This is why libTAS doesn't exist for windows. If you want to read up on windows general TAS software, and why it didn't work out, please have a read of this Hourglass post-mortem https://tasvideos.org/Forum/Topics/20210

**Reply by n0samu** [reply]
Here's something you all may be interested in: https://github.com/kmeisthax/flashtas

**Comment by p0008874** [comment, marked as answer]
I don't think the Elder will care about this anymore.
Even me.

**Reply by ghost** [reply]
No point reimplementing what libTAS already does.

**Reply by p0008874** [reply]
Whatever you say.
